Select peers on initial sync more carefully

UdjinM6 opened the pull request.

tgflynn commented on src/masternode-sync.cpp:
```diff
@@ -295,6 +295,27 @@ void CMasternodeSync::ProcessTick()
+            CNodeStateStats stats;
+            if(!GetNodeStateStats(pnode->id, stats) || stats.nCommonHeight == -1 || stats.nSyncHeight == -1) continue; // not enough info about this peer
+
+            if(pCurrentBlockIndex->nHeight > stats.nCommonHeight) {
```
Maybe it would be good to leave a little margin of error here.  If a new block just came in one node might receive it a little before the other.

UdjinM6 commented:
added small margins on both sides, fixed comment/log message to be more accurate (we don't really know if peer has headers/blocks until we get them, all we know is that peer claimed to have smth), rebased on top of latest 0.12.1.x

tgflynn commented:
utACK
